configs/configschema/schema.go

```diff
@@ -4,6 +4,13 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
+type StringKind int
+
+const (
+	StringPlain StringKind = iota
+	StringMarkdown
```

Comment:
Why are we introducing `StringMarkdown` here? Why does the schema need to know how the string will be interpreted?

Reply:
/cc @paultyng for this one.

Reply:
For a language server integration and for document generation for the registry (or anything else) we need the ability to store descriptions that are rich or we would need to duplicate efforts between web documentation and online documentation which seems problematic. This is the Langauge Server Specification that this model was taken from though: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#markupContent

Allowing providers to store descriptions in markdown would alleviate the need to also store attribute, block, and resource descriptions in documentation markdown files.